mruby/src/value/mod.rs

```diff
@@ -75,6 +80,32 @@ impl Value {
     }
 }
 
+impl Clone for Value {
+    fn clone(&self) -> Self {
+        let ruby_type = self.ruby_type();
+        if ruby_type == Data {
+            unsafe {
+                let ptr = mem::transmute::<*mut c_void, Rc<RefCell<_>>>(self.value.value.p);
+                let cloned_ptr = Rc::clone(&ptr);
+                Self::new(
+                    Rc::clone(&self.interp),
+                    sys::mrb_value {
+                        value: sys::mrb_value__bindgen_ty_1 {
+                            p: mem::transmute::<Rc<RefCell<_>>, *mut c_void>(cloned_ptr),
+                        },
+                        tt: self.value.tt
+                    },
+                )
+            }
+        } else {
+            Self::new(
+                Rc::clone(&self.interp),
```

Comment:
please rebase. I've changed the `Value` constructor to take an `&Mrb` to make this line less noisy.

Reply:
oh whoops. That might be still on a branch. nevermind 😅